File: app/src/main/java/org/demo/weatherapp/view/ViewAdapters.java
package org.demo.weatherapp.view;

import android.databinding.BindingAdapter;
import android.text.Html;
import android.widget.TextView;

public class ViewAdapters {

    private ViewAdapters() {
        //do nothing
    }

    @BindingAdapter("setWeatherIcon")
    public static void setWeatherIcon(TextView textView, String value) {
        if (value != null && !value.isEmpty()) {
            textView.setText(Html.fromHtml(value));
        }
    }
}

File: app/src/main/java/org/demo/weatherapp/view/ViewAdapters.kt
package org.demo.weatherapp.view

import android.databinding.BindingAdapter
import android.text.Html
import android.widget.TextView

class ViewAdapters {

    companion object {

        @JvmStatic
        @BindingAdapter("setWeatherIcon")
        fun setWeatherIcon(textView: TextView, value: String?) {
            if (value != null && !value.isEmpty()) {
                textView.text = Html.fromHtml(value)
            }
        }
    }
}
